cmake/get_info.cmake

```diff
@@ -0,0 +1,179 @@
+SET(detailed_log "${CMAKE_BINARY_DIR}/detailed.log")
+SET(minimal_log  "${CMAKE_BINARY_DIR}/minimal.log")
+FILE(REMOVE ${detailed_log} ${minimal_log})
+
+MACRO(_both)
+    # Write to both log files:
+    FILE(APPEND ${detailed_log} "${ARGN}")
+    FILE(APPEND ${minimal_log} "${ARGN}")
+ENDMACRO()
+
+MACRO(_detailed)
+    # Only write to detailed.log:
+    FILE(APPEND ${detailed_log} "${ARGN}")
+ENDMACRO()
+
+MACRO(_minimal)
+    # Only write to minimal.log:
+    FILE(APPEND ${minimal_log} "${ARGN}")
+ENDMACRO()
+
+function(ginkgo_print_generic_header log_type optional_string)
+    set(upd_string
+        "
+---------------------------------------------------------------------------------------------------------
+--
+--    ${optional_string}"
+        )
+    FILE(APPEND ${log_type} "${upd_string}")
+endfunction()
+
+function(ginkgo_print_module_header log_type module_name)
+    set(upd_string
+        "The ${module_name} module is being compiled.
+--
+--    CMake related ${module_name} module variables:"
+        )
+    ginkgo_print_generic_header(${log_type} "${upd_string}")
+endfunction()
+
+function(ginkgo_print_module_footer log_type optional_string)
+    set(upd_string
+        "
+--    ${optional_string}"
+        )
+    FILE(APPEND ${log_type} "${upd_string}")
+endfunction()
+
+FUNCTION(ginkgo_print_flags log_type var_name)
+    string(TOUPPER "${CMAKE_BUILD_TYPE}" suff)
+    set(var_string "${var_name}_${suff}")
+    if(${var_string} STREQUAL "")
+        set(str_value "<unspecified>")
+    else()
+        set(str_value "${${var_string}}")
+    endif()
+    string(SUBSTRING "
+--        ${var_string}:                                                        " 0 55 upd_string)
+    string(APPEND upd_string "${str_value}")
+    FILE(APPEND ${log_type} ${upd_string})
+ENDFUNCTION()
+
+function(ginkgo_print_variable log_type var_name)
+    string(SUBSTRING
+        "
+--        ${var_name}:                                                          " 0 55 upd_string)
+    if(${var_name} STREQUAL "")
```

Comment:
does it need to be "${${var_name}}"?

Reply:
I believe this should be correct, see [Variable Expansion](https://cmake.org/cmake/help/v3.18/command/if.html#variable-expansion)